File: primatis-web/src/app/auth/models/login-response.ts
/**
 * Contrat REST retourné par `POST /api/v1/auth/login` (voir
 * `be.primatis.security.web.LoginResponse` côté backend).
 *
 * `expiresAt` reste une chaîne ISO-8601 telle que reçue : elle n'est
 * convertie que si une interaction/affichage l'exige réellement.
 */
export interface LoginResponse {
  token: string;
  tokenType: string;
  expiresAt: string;
  expiresInSeconds: number;
}

File: primatis-api/src/main/java/be/primatis/security/web/LoginResponse.java
package be.primatis.security.web;

import be.primatis.security.AccessToken;

import java.time.Instant;

/**
 * Contrat REST retourné après une authentification réussie.
 *
 * Les rôles et permissions ne sont pas dupliqués dans cette réponse :
 * ils sont portés par les claims du JWT signé par le backend.
 */
public record LoginResponse(
        String token,
        String tokenType,
        Instant expiresAt,
        long expiresInSeconds) {

    public static LoginResponse from(AccessToken accessToken) {
        return new LoginResponse(
                accessToken.token(),
                accessToken.tokenType(),
                accessToken.expiresAt(),
                accessToken.expiresInSeconds());
    }
}
